dev-docs/tutorials/getting-started.md

```diff
@@ -0,0 +1,142 @@
+# Get started developing for Ubuntu Pro Client
+
+This tutorial will walk you through getting your development environment set up from scratch for contributing to Ubuntu Pro Client.
+
+We generally support the latest interim release and the latest LTS of Ubuntu as development hosts for Ubuntu Pro Client.
+
+> **Note**
+> This tutorial was last tested on Ubuntu 23.10 Mantic Minotaur (mantic).
+
+# From nothing to unit tests
+Open a terminal.
+
+Install necessary host tooling.
+```bash
+sudo apt install git tox pre-commit intltool libapt-pkg-dev sbuild-launchpad-chroot
+```
+> **Note**
+> Python comes installed by default on Ubuntu.
+> Do not install python via any non-Ubuntu means. That includes conda, pyenv, asdf, ppa:deadsnakes/ppa, download from python.org, etc.
+
+Get the code.
+```bash
+git clone https://github.com/canonical/ubuntu-pro-client.git
+cd ubuntu-pro-client
+```
+
+Initialize `pre-commit`.
+```bash
+pre-commit install
+```
+
+Run the unit tests and other checks.
+```bash
+tox
+```
+
+> **Note**
+> You can run individual jobs (or "environments" in tox terminology) with the `-e` flag.
+>
+> For example, `tox -e test` will only run the unit tests.
+>
+> Explore `tox.ini` to see what tox environments are available and how they're configured.
```

Comment:
Would it be beneficial to include additional examples? I'm torn because it would be nice to know right off the rip that `tox -e do_format` and `tox -e check_format` exists. 
(I never scrolled far enough down in tox.ini before to see that do_format exists so I've been doing all my formatting manually :smiling_face_with_tear:)

But yeah I think those 2 commands as well as `tox -e test` are essential for development and would be worth explicitly giving to the user. Because some new contributers will probably be like me: coming from a background of unfamiliarity with tox, and thus even if they were to read the whole tox.ini file, would not know what commands they actually have available to them.

Reply:
more than that, we could have a small section explaining style formatting, shellcheck, etc; all tox things deserve it

Reply:
do_format and check_format are cloud-init tox commands :), but I like the sentiment that we should list a couple more commands.

I'm wary of adding too much here though, I would rather encourage the reader to explore more on their own here, and have separate detailed docs for certain topics as necessary. But I'll try to add a short explanation of what we use tox for.

Reply:
And @a-dubs for ubuntu-pro-client I recommend setting up `pre-commit` as the tutorial recommends - that will format your code before committing.